### PR TITLE
Added EJS Language Support

### PR DESCRIPTION
**Pull request by steventhanna**
Added language support for the EJS templating language.

The EJS [project](https://github.com/tj/ejs) on GitHub has nearly 2,700 stars,  but if that is not enough qualification, here is the [search link](https://github.com/search?utf8=%E2%9C%93&q=ejs+extension%3Aejs&type=Code&ref=searchresults), in which over 40,000 files show up under the criteria `ejs extension:ejs`.

Samples have been added, and ACE Editor already has support for EJS.

### CODE REVIEW

**Comment by arfon** [comment]
Hi @steventhanna. Thanks for submitting this PR. Looking at the CI build failure it seems you've not actually included the submodule reference (see this [PR as an example](https://github.com/github/linguist/pull/2905/files)).

**Comment by pchaigno** [comment]
Looks like many of those files are not HTML though :/

**Comment by steventhanna** [comment]
One could argue that EJS is a system similar to Handlebars... But this disucssion has already taken place [here](https://github.com/github/linguist/pull/2625).  Counter argument to the first link is [here](https://github.com/github/linguist/pull/2586), but this PR was rejected in favor of the first link.

**Comment by arfon** [comment]
> Looks like many of those files are not HTML though :/

@pchaigno - could you link to some that don't appear to have any HTML in them?

**Comment by pchaigno** [comment]
Going through @steventhanna's [search example](https://github.com/search?utf8=%E2%9C%93&q=extension%3Aejs+NOT+html+NOT+include+NOT+li&type=Code&ref=searchresults), I found many files with no HTML or EJS syntax in them. All the files in the first page of results for instance. Tough, that probably does not make them invalid EJS files. I'm just not sure what they're supposed to be counted as...

**Comment by steventhanna** [comment]
Many times dev's create separate code "parts" that are later imported into templates for a more modular design, using the EJS syntax `<% include ... %>`, which would account for several files not having any EJS syntax at all.

> I found many files with no HTML or EJS syntax

@pchaigno Could you give a specific example of a file without any HTML?  All of the files that I have looked at have HTML characteristics and / or HTML tags.

**Comment by pchaigno** [comment]
[This one](https://github.com/peterlevel1/walkHTML/blob/2531e18908b08f1d1fbd5785db34d897ef4cfcb7/pkgs/ejs/test/res/1.ejs) for instance, or [this one](https://github.com/Arondight/arondight.github.io/blob/6678424e0de6549b21517596f7fb70be35ef8e0a/my/themes/hexo-theme-yelee/layout/_partial/comments/duoshuo.ejs).

**Comment by steventhanna** [comment]
Well, I stand corrected...  Not entirely sure how to go about fixing that, unless custom checks to the grammar submodule were added... Hopefully there is a better alternative.

File: vendor/grammars/ejs-tmbundle [hunk 1]
```diff
@@ -0,0 +1 @@
+Subproject commit 0854026c995c15cf7b02b6f20517258115f536d6
```
Comment (Glenntobby) [inline vendor/grammars/ejs-tmbundle]: #2353

**Comment by steventhanna** [comment]
@arfon and @pchaigno, you guys seem to have great influence over this repo, any suggestions for getting this PR merged?

**Comment by arfon** [comment]
> Well, I stand corrected... Not entirely sure how to go about fixing that, unless custom checks to the grammar submodule were added... Hopefully there is a better alternative.

All of these files that I've checked still appear to be `EJS` files so I think we're good.

Closing in favour of #2993

**Pull request closed without merging**